### PR TITLE
fix #128 - add oracle support

### LINKED ISSUE #128: Request: add Oracle DB support
I checked the created issues and I didn't realize some issue is related to this, so I'm creating it.

As stated in Moodle Dev Telegram chat, It would be really useful to support Oracle (this issue) and mssql (issue #92).

I've tried adding Oracle support in one plugin and then checked that only mysql, pgsql and mariadb are supported db-types.

As I can see, I think it is necessary mainly to add an implementation class of any db-type under https://github.com/moodlehq/moodle-plugin-ci/tree/master/src/Installer/Database.

My original idea was to use the docker image for Oracle used under moodlehq/moodle-docker. However, Github Actions require public docker hub images (if I understand it correct) and it seems it isn't. So, I was trying another db image, gvenzl/oracle-xe (supporting 18c and 11 Oracle versions). I successfully managed to build the image, but then I realized moodle-plugin-ci does not support it.

My last trials can be found [here](https://github.com/jpahullo/moodle-tool_mergeusers/runs/3285053741?check_suite_focus=true). One example is this:

image

One really useful point of that oracle image is that is has a builtin checker for db being ready. It takes a bit of time however:

image

The important bits of the yml are the following:

```
    services:
      oracle:
        image: gvenzl/oracle-xe
        env:
          ORACLE_RANDOM_PASSWORD: yes
          ORACLE_DATABASE: moodle
          APP_USER: moodle
          APP_USER_PASSWORD: moodle
        ports:
          - 1521:1521
        options: >-
          --health-cmd healthcheck.sh
          --health-interval 20s
          --health-timeout 10s
          --health-retries 10
```

This image requires non-empty passwords, so I needed to set up non-empty passwords on the ci installing:

```
      - name: Install moodle-plugin-ci
        run: |
          moodle-plugin-ci install --plugin ./plugin --db-host=127.0.0.1 --db-user moodle --db-pass moodle --db-name moodle
        env:
          DB: ${{ matrix.database }}
          MOODLE_BRANCH: ${{ matrix.moodle-branch }}
```

### PR DESCRIPTION
**Pull request by jpahullo**
Use moodlehq/moodle-db-oracle as oracle backend.
For performance, we alternate db type testing for
every test included.

### CODE REVIEW

**Comment by jpahullo** [comment]
I'm creating the PR to be reviewed its code. I know there is some tests failing,  but due to a bug in Moodle core. It can be seen at https://tracker.moodle.org/browse/MDL-72318

File: src/Installer/Database/OracleDatabase.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Moodle Plugin CI package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Copyright (c) 2018 Blackboard Inc. (http://www.blackboard.com)
```
Comment (kabalin) [inline src/Installer/Database/OracleDatabase.php:9]: Please change this block according to #131

File: tests/Installer/Database/OracleDatabaseTest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Moodle Plugin CI package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Copyright (c) 2018 Blackboard Inc. (http://www.blackboard.com)
```
Comment (kabalin) [inline tests/Installer/Database/OracleDatabaseTest.php:9]: Please change this block according to #131

**Review by kabalin — COMMENTED** [review]
Hi @jpahullo, thanks for working on this. I left some comments that should help you with making this ready for merging. Also you would need to amend default yml files for travis and github.

**Comment by stronk7** [comment]
Hi,

as far as this is quite old and Oracle is being deprecated and will be removed in a couple of versions, I am going to close this now.

Reference: https://moodle.org/mod/forum/discuss.php?d=456122

**Pull request closed without merging**